### PR TITLE
refactor(share/p2p/shrex): use default timeout for shrex-eds client 

### PR DESCRIPTION
**Pull request by walldiss**
## Overview
Use default timeout for shrex-eds client if no deadline specified by provided context.

### CODE REVIEW

File: share/p2p/shrexeds/client.go [hunk 1]
```diff
@@ -131,3 +131,28 @@ func (c *Client) doRequest(
...
+	// if deadline not set, client read deadline defaults to server write deadline
+	if c.params.ServerWriteTimeout != 0 {
+		err := stream.SetReadDeadline(time.Now().Add(c.params.ServerWriteTimeout))
+		if err != nil {
+			log.Debugw("client: setting read deadline", "err", err)
+		}
+	}
+
+	// if deadline not set, client write deadline defaults to server read deadline
+	if c.params.ServerReadTimeout != 0 {
```
Comment (Wondertan) [inline share/p2p/shrexeds/client.go:152]: Server timeouts on the client?

Comment (distractedm1nd) [reply share/p2p/shrexeds/client.go:152]: This is how its done in shrexnd. It makes sense, maybe we could add some constant for the actual networking though

File: share/p2p/shrexeds/client.go [hunk 2]
```diff
@@ -131,3 +131,28 @@ func (c *Client) doRequest(
 		return nil, p2p.ErrInvalidResponse
 	}
 }
+
+func (c *Client) setStreamDeadlines(ctx context.Context, stream network.Stream) {
+	// set read/write deadline to use context deadline if it exists
+	if dl, ok := ctx.Deadline(); ok {
+		if err := stream.SetDeadline(dl); err != nil {
+			log.Debugw("client: setting deadline: %s", "err", err)
+		}
```
Comment (renaynay) [inline share/p2p/shrexeds/client.go:140]:
```suggestion
		if err := stream.SetDeadline(dl); err == nil {
		      return
		}
		log.Debugw("client: setting deadline: %s", "err", err)
```

File: share/p2p/shrexeds/client.go (hunk 1 above)
Comment (walldiss) [reply share/p2p/shrexeds/client.go:152]: This is setting same timeout for the other side of pipe. We could do separate client side timeouts later if we would need, but as for default settings those seems reasonable to mirror server side.

File: share/p2p/shrexeds/client.go (hunk 2 above)
Comment (walldiss) [reply share/p2p/shrexeds/client.go]: It should be more like:
```
if err == nil{
   return
}
log.Error
```
Nice catch!

**Review by Wondertan — DISMISSED** [review]

File: share/p2p/shrexnd/client.go [hunk 3]
```diff
@@ -160,10 +160,10 @@ func (c *Client) setStreamDeadlines(ctx context.Context, stream network.Stream)
 	deadline, ok := ctx.Deadline()
 	if ok {
 		err := stream.SetDeadline(deadline)
-		if err != nil {
-			log.Debugw("client-nd: set write deadline", "err", err)
+		if err == nil {
+			return
 		}
-		return
+		log.Debugw("client-nd: set write deadline", "err", err)
```
Comment (renaynay) [inline share/p2p/shrexnd/client.go:166]:
```suggestion
		log.Debugw("client-nd: set stream deadline", "err", err)
```

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**